examples/calibration/plot_calibration_curve.py

```diff
@@ -5,131 +5,300 @@
+# monotonic transformation. Indeed, no rank metrics are affected by
+# calibration.
 
-    # Logistic regression with no calibration as baseline
-    lr = LogisticRegression(C=1.)
+index = []
+brier = []
+logloss = []
+precision = []
+recall = []
+f1 = []
+roc_auc = []
 
-    fig = plt.figure(fig_index, figsize=(10, 10))
-    ax1 = plt.subplot2grid((3, 1), (0, 0), rowspan=2, fig=fig)
-    ax2 = plt.subplot2grid((3, 1), (2, 0), fig=fig)
+for i, (clf, name) in enumerate(clf_list):
+    clf.fit(X_train, y_train)
+    y_proba = clf.predict_proba(X_test)
+    y_pred = clf.predict(X_test)
 
-    ax1.plot([0, 1], [0, 1], "k:", label="Perfectly calibrated")
-    for clf, name in [(lr, 'Logistic'),
-                      (est, name),
-                      (isotonic, name + ' + Isotonic'),
-                      (sigmoid, name + ' + Sigmoid')]:
-        clf.fit(X_train, y_train)
-        y_pred = clf.predict(X_test)
-        if hasattr(clf, "predict_proba"):
-            prob_pos = clf.predict_proba(X_test)[:, 1]
-        else:  # use decision function
-            prob_pos = clf.decision_function(X_test)
-            prob_pos = \
-                (prob_pos - prob_pos.min()) / (prob_pos.max() - prob_pos.min())
+    # Create DataFrame index
+    index.append(name)
+    # Store column data
+    brier.append(brier_score_loss(y_test, y_proba[:, 1]))
+    logloss.append(log_loss(y_test, y_proba[:, 1]))
+    precision.append(precision_score(y_test, y_pred))
+    recall.append(recall_score(y_test, y_pred))
+    f1.append(f1_score(y_test, y_pred))
+    roc_auc.append(roc_auc_score(y_test, y_pred))
 
-        clf_score = brier_score_loss(y_test, prob_pos, pos_label=y.max())
-        print("%s:" % name)
-        print("\tBrier: %1.3f" % (clf_score))
-        print("\tPrecision: %1.3f" % precision_score(y_test, y_pred))
-        print("\tRecall: %1.3f" % recall_score(y_test, y_pred))
-        print("\tF1: %1.3f\n" % f1_score(y_test, y_pred))
+score_df = pd.DataFrame(
+    data={'Brier loss': brier, 'Log loss': logloss, 'Precision': precision,
+          'Recall': recall, 'F1': f1, 'ROC AUC': roc_auc},
+    index=index,
+)
+score_df.round(3)
 
-        fraction_of_positives, mean_predicted_value = \
-            calibration_curve(y_test, prob_pos, n_bins=10)
+# %%
+# Next, we will compare:
+#
+# * :class:`~sklearn.linear_model.LogisticRegression` (baseline)
+# * Uncalibrated :class:`~sklearn.svm.LinearSVC`. Since SVC does not output
+#   probabilities by default, we naively scale the output of the
+#   :term:`decision_function` into [0, 1] by applying min-max scaling.
+# * :class:`~sklearn.svm.LinearSVC` with isotonic and sigmoid
+#   calibration (see :ref:`User Guide <calibration>`)
+#
 
-        ax1.plot(mean_predicted_value, fraction_of_positives, "s-",
-                 label="%s (%1.3f)" % (name, clf_score))
 
-        ax2.hist(prob_pos, range=(0, 1), bins=10, label=name,
-                 histtype="step", lw=2)
+class NaivelyCalibratedLinearSVC(LinearSVC):
```

Comment:
I would create 3 cells: the naive class on its own, the model declarations in another, and then the plotting.